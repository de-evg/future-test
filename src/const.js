export const showingStatus = {
    UNSET: `UNSET`,
    SHOW: `SHOW`,
    HIDE: `HIDE`
};

export const DEFAULT_SORT = `ID_UP`;

export const SortType = {
    ID_UP: `ID_UP`,
    ID_DOWN: `ID_DOWN`,
    FIRSTNAME_UP: `FIRSTNAME_UP`,
    FIRSTNAME_DOWN: `FIRSTNAME_DOWN`,
    LASTNAME_UP: `LASTNAME_UP`,
    LASTNAME_DOWN: `LASTNAME_DOWN`,
    EMAIL_UP: `EMAIL_UP`,
    EMAIL_DOWN: `EMAIL_DOWN`,
    PHONE_UP: `PHONE_UP`,
    PHONE_DOWN: `PHONE_DOWN`,
};