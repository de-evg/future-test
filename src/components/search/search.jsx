import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {NameSpace} from '../../store/reducers/root';
import {showingStatus} from '../../const';

const Search = ({searchFormVisualStatus}) => {
  let visualClass = "";
  if (searchFormVisualStatus === showingStatus.HIDE) {
    visualClass = "search-form--hidden";
  }
  if (searchFormVisualStatus === showingStatus.SHOW) {
    visualClass = "search-form--show";
  }
  return (
    <form className={`search-form ${visualClass}`} action="#">
      <input id="search" className="search-input" type="text" />
      <button className="search-button" type="submit">Найти</button>
    </form>
  );
};

Search.propTypes = {
  searchFormVisualStatus: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  searchFormVisualStatus: state[NameSpace.INTERFACE_DATA].searchFormVisualStatus
});

export default connect(mapStateToProps)(Search);
