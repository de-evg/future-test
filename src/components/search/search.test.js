import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { Search } from "./search";
import Adapter from "enzyme-adapter-react-16";
import {showingStatus} from "../../const";

Enzyme.configure({ adapter: new Adapter() });

describe(`Render Search correctly`, () => {
  it(`Reder Search with UNSET showing status correctly`, () => {
    const wrapper = shallow(<Search searchFormVisualStatus={showingStatus.UNSET} setFilter={() => {}}  />);
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it(`Reder Search with SHOW showing status correctly`, () => {
    const wrapper = shallow(<Search searchFormVisualStatus={showingStatus.SHOW} setFilter={() => {}}  />);
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it(`Reder Search with HIDE showing status correctly`, () => {
    const wrapper = shallow(<Search searchFormVisualStatus={showingStatus.HIDE} setFilter={() => {}}  />);
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
