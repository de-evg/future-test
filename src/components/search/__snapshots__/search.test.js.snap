// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Search correctly Reder Search with HIDE showing status correctly 1`] = `
<form
  action="#"
  className="search-form search-form--hidden"
  onSubmit={[Function]}
>
  <input
    className="search-input"
    id="filter"
    onChange={[Function]}
    type="text"
  />
  <button
    className="search-button"
    type="submit"
  >
    Найти
  </button>
</form>
`;

exports[`Render Search correctly Reder Search with SHOW showing status correctly 1`] = `
<form
  action="#"
  className="search-form search-form--show"
  onSubmit={[Function]}
>
  <input
    className="search-input"
    id="filter"
    onChange={[Function]}
    type="text"
  />
  <button
    className="search-button"
    type="submit"
  >
    Найти
  </button>
</form>
`;

exports[`Render Search correctly Reder Search with UNSET showing status correctly 1`] = `
<form
  action="#"
  className="search-form "
  onSubmit={[Function]}
>
  <input
    className="search-input"
    id="filter"
    onChange={[Function]}
    type="text"
  />
  <button
    className="search-button"
    type="submit"
  >
    Найти
  </button>
</form>
`;
